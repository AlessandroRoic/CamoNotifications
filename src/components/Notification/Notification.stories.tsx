import { ComponentMeta, ComponentStory } from '@storybook/react';
import React from 'react';
import { Notification, NotificationProps, NotificationType } from './index';

export default {
  title: 'Notification',
  component: Notification,
} as ComponentMeta<typeof Notification>;

const Template: ComponentStory<typeof Notification> = (props: NotificationProps) => {
  return <Notification {...props} />;
};

export const Cookie = Template.bind({});
Cookie.args = {
  id: 'cookie',
  type: NotificationType.COOKIE,
  title: 'Warning',
  isOpen: true,
  onClose: () => {},
  subtitle: 'Accept Cookies',
  content: 'Click here to accept cookies:',
};
