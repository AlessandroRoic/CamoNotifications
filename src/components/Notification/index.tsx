export { Notification, NotificationType } from './Notification';
export type { NotificationProps } from './Notification';
