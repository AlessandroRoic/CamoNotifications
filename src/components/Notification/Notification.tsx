import React, { useEffect, useState } from 'react';
import ReactDOM from 'react-dom';
import { HarmonyConfig, HarmonyType, NotificationProps, Position } from './Notification.types';
import { CloseButtonWrapper, NotificationWrapper, StyledNotification } from './Notification.style';

export default function Notification(props: NotificationProps): React.ReactElement | null {
  const { id, notificationType, position, harmonyType, title, subtitle, content, customStyle, isOpen, onClose } = props;

  const [notificationStyle, setNotificationStyle]: [React.CSSProperties, (value: React.CSSProperties) => void] = useState({});

  const parseRGBToArray = (rgb: string): number[] =>
    rgb
      .replace(/[^\d,]/g, '')
      .split(',')
      .map(Number);

  /**
   * Converts an RGB color value to HSL. Conversion formula
   * adapted from https://en.wikipedia.org/wiki/HSL_and_HSV#Hue_and_chroma.
   * Assumes r, g, and b are contained in the set [0, 255] and
   * returns HLS with H: [0, 360], S: [0, 100], L: [0, 100]
   *
   * @param   rgb The rgb string from the style
   * @return  The HSL representation
   */
  const parseRGBtoHSL = (rgb: string): number[] => {
    let [r, g, b] = parseRGBToArray(rgb);
    r /= 255;
    g /= 255;
    b /= 255;
    const min = Math.min(r, g, b);
    const max = Math.max(r, g, b);
    const delta = max - min;
    let [h, s, l] = [0, 0, 0];
    if (delta === 0) {
      h = 0;
    } else if (max === r) {
      h = ((g - b) / delta) % 6;
    } else if (max === g) {
      h = (b - r) / delta + 2;
    } else {
      h = (r - g) / delta + 4;
    }
    h = Math.round(h * 60);
    h = h < 0 ? (h += 360) : h;
    l = (max + min) / 2;
    s = delta === 0 ? 0 : delta / (1 - Math.abs(2 * l - 1));
    s = +(s * 100).toFixed(1);
    l = +(l * 100).toFixed(1);
    return [h, s, l];
  };

  const getHarmonyConfig = (harmony?: HarmonyType): HarmonyConfig => {
    switch (harmony) {
      case HarmonyType.SPLIT:
        return { start: 150, end: 210, interval: 1 };
      case HarmonyType.TRIAD:
        return { start: 120, end: 240, interval: 120 };
      case HarmonyType.TETRAD:
        return { start: 90, end: 270, interval: 90 };
      case HarmonyType.ANALOGOUS:
        return { start: 30, end: 90, interval: 30 };
      case HarmonyType.COMPLIMENTARY:
      default:
        return { start: 180, end: 180, interval: 1 };
    }
  };

  /**
   * Formula adapted from https://en.wikipedia.org/wiki/Relative_luminance
   * @input selected color expressed in RBG
   * @return relative luminance of an RGB color
   */
  const luminance = (r: number, g: number, b: number): number => {
    const mappedRgb = [r, g, b].map((color: number) => {
      const parsedColor = color / 255;
      return parsedColor <= 0.03928 ? parsedColor / 12.92 : ((parsedColor + 0.055) / 1.055) ** 2.4;
    });
    return mappedRgb[0] * 0.2126 + mappedRgb[1] * 0.7152 + mappedRgb[2] * 0.0722;
  };

  /** Calculates the contrast between two colors
   * @input colors to check
   * @return contrast between two colors
   */
  const contrast = (rgb1: number[], rgb2: number[]): number => {
    const lum1 = luminance(rgb1[0], rgb1[1], rgb1[2]);
    const lum2 = luminance(rgb2[0], rgb2[1], rgb2[2]);
    const brightest = Math.max(lum1, lum2);
    const darkest = Math.min(lum1, lum2);
    return (brightest + 0.05) / (darkest + 0.05);
  };

  const getColorHarmony = (rgba: string, harmony?: HarmonyType): string[] => {
    const [h, s, l] = parseRGBtoHSL(rgba);
    const colors = [];
    const { start, end, interval } = getHarmonyConfig(harmony);

    for (let i = start; i <= end; i += interval) {
      const harmonizedH = (h + i) % 255;
      const c1 = `hsl(${harmonizedH}, ${s}%, ${l}%)`;
      colors.push(c1);
    }
    return colors;
  };

  const white = [255, 255, 255];

  const getFontColor = (backgroundColor: any): string => {
    const backgroundContrast = contrast(parseRGBToArray(backgroundColor), white);
    return backgroundContrast >= 4.5 ? 'rgb(255,255,255)' : 'rgb(0,0,0)';
  };

  useEffect(() => {
    const { backgroundColor } = window.getComputedStyle(document.body);
    const generatedStyle = {
      color: getFontColor(backgroundColor),
      backgroundColor: getColorHarmony(backgroundColor, harmonyType)[0],
    };

    setNotificationStyle({
      ...generatedStyle,
      ...(customStyle || {}),
      zIndex: customStyle?.zIndex || 9999,
    });
  }, [isOpen]);

  if (!isOpen) return null;
  return ReactDOM.createPortal(
    <NotificationWrapper position={position} id={id}>
      <StyledNotification notificationType={notificationType} style={notificationStyle}>
        <CloseButtonWrapper>
          <button type="button" onClick={onClose} style={{ fill: notificationStyle.color }}>
            <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24">
              <path d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z" />
            </svg>
          </button>
        </CloseButtonWrapper>
        {title && <h2>{title}</h2>}
        {subtitle && <h3>{subtitle}</h3>}
        {content && <span>{content}</span>}
      </StyledNotification>
    </NotificationWrapper>,
    document.body,
  );
}

Notification.defaultProps = {
  title: undefined,
  subtitle: undefined,
  content: undefined,
  customStyle: undefined,
  position: Position.CENTER,
  harmonyType: HarmonyType.COMPLIMENTARY,
};
