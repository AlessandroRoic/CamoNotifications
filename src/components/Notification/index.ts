import Notification from './Notification';

export { NotificationType, Position, HarmonyType } from './Notification.types';
export type { NotificationProps, HarmonyConfig } from './Notification.types';
export { Notification };
