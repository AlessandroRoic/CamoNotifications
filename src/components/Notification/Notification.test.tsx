import React, { useState } from 'react';
import { cleanup, fireEvent, render } from '@testing-library/react';
import Notification from './Notification';
import { HarmonyType, NotificationProps, NotificationType, Position } from './Notification.types';
import { getPosition, getWidth } from './Notification.style';

const defaultProps: NotificationProps = { id: 'test', isOpen: true, onClose: () => {}, notificationType: NotificationType.MEDIUM };
const customRender = (props: NotificationProps = defaultProps) => render(<Notification {...props} />);

function TestWrapper(props: { initialState: any } = { initialState: true }) {
  const { initialState } = props;
  const [isOpen, setIsOpen] = useState(initialState);
  return <Notification id="test" isOpen={isOpen} onClose={() => setIsOpen(false)} notificationType={NotificationType.MEDIUM} />;
}

describe('notification component', () => {
  beforeEach(cleanup);

  it('should render correctly', () => {
    const component = customRender({ ...defaultProps, title: 'test', subtitle: 'test', content: 'test' });
    expect(document.getElementById('test')).toBeTruthy();
    expect(component.baseElement).toMatchSnapshot();
  });

  it('should close correctly with button', () => {
    const component = render(<TestWrapper initialState />);
    fireEvent.click(component.getByRole('button'));
    expect(document.getElementById('test')).toBeNull();
    expect(component.baseElement).toMatchSnapshot();
  });

  it.each(Object.values(NotificationType))('should render notification type correctly', (notificationType: NotificationType) => {
    const component = customRender({ ...defaultProps, notificationType });
    const wrapper = document.getElementById('test')?.firstChild;
    const generatedWidthByType = { width: getWidth(notificationType) };
    expect(wrapper).toHaveStyle(generatedWidthByType);
    expect(component.baseElement).toMatchSnapshot();
  });

  it.each(Object.values(Position))('should render position correctly', (position: Position) => {
    const component = customRender({ ...defaultProps, position });
    const wrapper = document.getElementById('test');
    const generatedPosition = getPosition(position);
    expect(wrapper).toHaveStyle(generatedPosition);
    expect(component.baseElement).toMatchSnapshot();
  });

  it.each(Object.values(HarmonyType))('should render harmony correctly', (harmonyType: HarmonyType) => {
    const component = customRender({ ...defaultProps, harmonyType });
    expect(component.baseElement).toMatchSnapshot();
  });

  it('should render no harmony', () => {
    const component = customRender({ ...defaultProps, harmonyType: undefined });
    expect(component.baseElement).toMatchSnapshot();
  });

  it('should render a white background notification', () => {
    document.body.style.background = 'rgb(0,0,0)';
    const component = customRender();
    expect(component.baseElement).toMatchSnapshot();
  });
});
