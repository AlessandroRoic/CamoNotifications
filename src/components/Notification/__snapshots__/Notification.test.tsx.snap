// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`notification component should close correctly with button 1`] = `
<body>
  <div />
</body>
`;

exports[`notification component should render a white background notification 1`] = `
<body
  style="background: rgb(0, 0, 0);"
>
  <div />
  <div
    class="sc-bczRLJ fIZRfA"
    id="test"
  >
    <div
      class="sc-gsnTZi bHbWND"
      style="color: rgb(255, 255, 255); background-color: rgb(0, 0, 0); z-index: 9999;"
    >
      <div
        class="sc-dkzDqf kLgSvk"
      >
        <button
          style="fill: rgb(255,255,255);"
          type="button"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`notification component should render correctly 1`] = `
<body>
  <div />
  <div
    class="sc-bczRLJ fIZRfA"
    id="test"
  >
    <div
      class="sc-gsnTZi bHbWND"
      style="color: rgb(0, 0, 0); z-index: 9999;"
    >
      <div
        class="sc-dkzDqf kLgSvk"
      >
        <button
          style="fill: rgb(0,0,0);"
          type="button"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
            />
          </svg>
        </button>
      </div>
      <h2>
        test
      </h2>
      <h3>
        test
      </h3>
      <span>
        test
      </span>
    </div>
  </div>
</body>
`;

exports[`notification component should render harmony correctly 1`] = `
<body>
  <div />
  <div
    class="sc-bczRLJ fIZRfA"
    id="test"
  >
    <div
      class="sc-gsnTZi bHbWND"
      style="color: rgb(0, 0, 0); z-index: 9999;"
    >
      <div
        class="sc-dkzDqf kLgSvk"
      >
        <button
          style="fill: rgb(0,0,0);"
          type="button"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`notification component should render harmony correctly 2`] = `
<body>
  <div />
  <div
    class="sc-bczRLJ fIZRfA"
    id="test"
  >
    <div
      class="sc-gsnTZi bHbWND"
      style="color: rgb(0, 0, 0); z-index: 9999;"
    >
      <div
        class="sc-dkzDqf kLgSvk"
      >
        <button
          style="fill: rgb(0,0,0);"
          type="button"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`notification component should render harmony correctly 3`] = `
<body>
  <div />
  <div
    class="sc-bczRLJ fIZRfA"
    id="test"
  >
    <div
      class="sc-gsnTZi bHbWND"
      style="color: rgb(0, 0, 0); z-index: 9999;"
    >
      <div
        class="sc-dkzDqf kLgSvk"
      >
        <button
          style="fill: rgb(0,0,0);"
          type="button"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`notification component should render harmony correctly 4`] = `
<body>
  <div />
  <div
    class="sc-bczRLJ fIZRfA"
    id="test"
  >
    <div
      class="sc-gsnTZi bHbWND"
      style="color: rgb(0, 0, 0); z-index: 9999;"
    >
      <div
        class="sc-dkzDqf kLgSvk"
      >
        <button
          style="fill: rgb(0,0,0);"
          type="button"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`notification component should render harmony correctly 5`] = `
<body>
  <div />
  <div
    class="sc-bczRLJ fIZRfA"
    id="test"
  >
    <div
      class="sc-gsnTZi bHbWND"
      style="color: rgb(0, 0, 0); z-index: 9999;"
    >
      <div
        class="sc-dkzDqf kLgSvk"
      >
        <button
          style="fill: rgb(0,0,0);"
          type="button"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`notification component should render harmony correctly 6`] = `
<body>
  <div />
  <div
    class="sc-bczRLJ fIZRfA"
    id="test"
  >
    <div
      class="sc-gsnTZi bHbWND"
      style="color: rgb(0, 0, 0); z-index: 9999;"
    >
      <div
        class="sc-dkzDqf kLgSvk"
      >
        <button
          style="fill: rgb(0,0,0);"
          type="button"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`notification component should render harmony correctly 7`] = `
<body>
  <div />
  <div
    class="sc-bczRLJ fIZRfA"
    id="test"
  >
    <div
      class="sc-gsnTZi bHbWND"
      style="color: rgb(0, 0, 0); z-index: 9999;"
    >
      <div
        class="sc-dkzDqf kLgSvk"
      >
        <button
          style="fill: rgb(0,0,0);"
          type="button"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`notification component should render harmony correctly 8`] = `
<body>
  <div />
  <div
    class="sc-bczRLJ fIZRfA"
    id="test"
  >
    <div
      class="sc-gsnTZi bHbWND"
      style="color: rgb(0, 0, 0); z-index: 9999;"
    >
      <div
        class="sc-dkzDqf kLgSvk"
      >
        <button
          style="fill: rgb(0,0,0);"
          type="button"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`notification component should render harmony correctly 9`] = `
<body>
  <div />
  <div
    class="sc-bczRLJ fIZRfA"
    id="test"
  >
    <div
      class="sc-gsnTZi bHbWND"
      style="color: rgb(0, 0, 0); z-index: 9999;"
    >
      <div
        class="sc-dkzDqf kLgSvk"
      >
        <button
          style="fill: rgb(0,0,0);"
          type="button"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`notification component should render harmony correctly 10`] = `
<body>
  <div />
  <div
    class="sc-bczRLJ fIZRfA"
    id="test"
  >
    <div
      class="sc-gsnTZi bHbWND"
      style="color: rgb(0, 0, 0); z-index: 9999;"
    >
      <div
        class="sc-dkzDqf kLgSvk"
      >
        <button
          style="fill: rgb(0,0,0);"
          type="button"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`notification component should render no harmony 1`] = `
<body>
  <div />
  <div
    class="sc-bczRLJ fIZRfA"
    id="test"
  >
    <div
      class="sc-gsnTZi bHbWND"
      style="color: rgb(0, 0, 0); z-index: 9999;"
    >
      <div
        class="sc-dkzDqf kLgSvk"
      >
        <button
          style="fill: rgb(0,0,0);"
          type="button"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`notification component should render notification type correctly 1`] = `
<body>
  <div />
  <div
    class="sc-bczRLJ fIZRfA"
    id="test"
  >
    <div
      class="sc-gsnTZi emVcHX"
      style="color: rgb(0, 0, 0); z-index: 9999;"
    >
      <div
        class="sc-dkzDqf kLgSvk"
      >
        <button
          style="fill: rgb(0,0,0);"
          type="button"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`notification component should render notification type correctly 2`] = `
<body>
  <div />
  <div
    class="sc-bczRLJ fIZRfA"
    id="test"
  >
    <div
      class="sc-gsnTZi kKeVcZ"
      style="color: rgb(0, 0, 0); z-index: 9999;"
    >
      <div
        class="sc-dkzDqf kLgSvk"
      >
        <button
          style="fill: rgb(0,0,0);"
          type="button"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`notification component should render notification type correctly 3`] = `
<body>
  <div />
  <div
    class="sc-bczRLJ fIZRfA"
    id="test"
  >
    <div
      class="sc-gsnTZi cFnQBZ"
      style="color: rgb(0, 0, 0); z-index: 9999;"
    >
      <div
        class="sc-dkzDqf kLgSvk"
      >
        <button
          style="fill: rgb(0,0,0);"
          type="button"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`notification component should render notification type correctly 4`] = `
<body>
  <div />
  <div
    class="sc-bczRLJ fIZRfA"
    id="test"
  >
    <div
      class="sc-gsnTZi bHbWND"
      style="color: rgb(0, 0, 0); z-index: 9999;"
    >
      <div
        class="sc-dkzDqf kLgSvk"
      >
        <button
          style="fill: rgb(0,0,0);"
          type="button"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`notification component should render notification type correctly 5`] = `
<body>
  <div />
  <div
    class="sc-bczRLJ fIZRfA"
    id="test"
  >
    <div
      class="sc-gsnTZi kKeVcZ"
      style="color: rgb(0, 0, 0); z-index: 9999;"
    >
      <div
        class="sc-dkzDqf kLgSvk"
      >
        <button
          style="fill: rgb(0,0,0);"
          type="button"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`notification component should render position correctly 1`] = `
<body>
  <div />
  <div
    class="sc-bczRLJ linwii"
    id="test"
  >
    <div
      class="sc-gsnTZi bHbWND"
      style="color: rgb(0, 0, 0); z-index: 9999;"
    >
      <div
        class="sc-dkzDqf kLgSvk"
      >
        <button
          style="fill: rgb(0,0,0);"
          type="button"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`notification component should render position correctly 2`] = `
<body>
  <div />
  <div
    class="sc-bczRLJ hfpaCx"
    id="test"
  >
    <div
      class="sc-gsnTZi bHbWND"
      style="color: rgb(0, 0, 0); z-index: 9999;"
    >
      <div
        class="sc-dkzDqf kLgSvk"
      >
        <button
          style="fill: rgb(0,0,0);"
          type="button"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`notification component should render position correctly 3`] = `
<body>
  <div />
  <div
    class="sc-bczRLJ cBnEAR"
    id="test"
  >
    <div
      class="sc-gsnTZi bHbWND"
      style="color: rgb(0, 0, 0); z-index: 9999;"
    >
      <div
        class="sc-dkzDqf kLgSvk"
      >
        <button
          style="fill: rgb(0,0,0);"
          type="button"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`notification component should render position correctly 4`] = `
<body>
  <div />
  <div
    class="sc-bczRLJ fIZRfA"
    id="test"
  >
    <div
      class="sc-gsnTZi bHbWND"
      style="color: rgb(0, 0, 0); z-index: 9999;"
    >
      <div
        class="sc-dkzDqf kLgSvk"
      >
        <button
          style="fill: rgb(0,0,0);"
          type="button"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`notification component should render position correctly 5`] = `
<body>
  <div />
  <div
    class="sc-bczRLJ bYGgfI"
    id="test"
  >
    <div
      class="sc-gsnTZi bHbWND"
      style="color: rgb(0, 0, 0); z-index: 9999;"
    >
      <div
        class="sc-dkzDqf kLgSvk"
      >
        <button
          style="fill: rgb(0,0,0);"
          type="button"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`notification component should render position correctly 6`] = `
<body>
  <div />
  <div
    class="sc-bczRLJ elUkKT"
    id="test"
  >
    <div
      class="sc-gsnTZi bHbWND"
      style="color: rgb(0, 0, 0); z-index: 9999;"
    >
      <div
        class="sc-dkzDqf kLgSvk"
      >
        <button
          style="fill: rgb(0,0,0);"
          type="button"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`notification component should render position correctly 7`] = `
<body>
  <div />
  <div
    class="sc-bczRLJ fA-dejy"
    id="test"
  >
    <div
      class="sc-gsnTZi bHbWND"
      style="color: rgb(0, 0, 0); z-index: 9999;"
    >
      <div
        class="sc-dkzDqf kLgSvk"
      >
        <button
          style="fill: rgb(0,0,0);"
          type="button"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`notification component should render position correctly 8`] = `
<body>
  <div />
  <div
    class="sc-bczRLJ hHtsR"
    id="test"
  >
    <div
      class="sc-gsnTZi bHbWND"
      style="color: rgb(0, 0, 0); z-index: 9999;"
    >
      <div
        class="sc-dkzDqf kLgSvk"
      >
        <button
          style="fill: rgb(0,0,0);"
          type="button"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`notification component should render position correctly 9`] = `
<body>
  <div />
  <div
    class="sc-bczRLJ jYcvoV"
    id="test"
  >
    <div
      class="sc-gsnTZi bHbWND"
      style="color: rgb(0, 0, 0); z-index: 9999;"
    >
      <div
        class="sc-dkzDqf kLgSvk"
      >
        <button
          style="fill: rgb(0,0,0);"
          type="button"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</body>
`;
